examples/nemo_ppo_inference.py

```diff
@@ -0,0 +1,78 @@
+import os.path
+import sys
+from glob import glob
+
+from nemo.collections.nlp.modules.common.megatron.megatron_init import (
+    fake_initialize_model_parallel,
+)
+from nemo.utils.app_state import AppState
+from omegaconf.omegaconf import OmegaConf
+
+from trlx.data.default_configs import default_ppo_config
+from trlx.trainer.nemo_ppo_trainer import PPOGPT, megatron_trainer
+
+default_config = default_ppo_config()
+
+trl_config = default_config.evolve(
+    train=dict(
+        default_config.train.__dict__,
+        trainer="NeMoPPOTrainer",
+        trainer_kwargs=dict(
+            pretrained_model=None,
+            megatron_cfg="megatron_20b.yaml",
+        ),
+    ),
+)
+
+
+def find_checkpoints(checkpoint_dir):
+    checkpoints = glob(os.path.join(checkpoint_dir, "*", "*.ckpt"))
+    names = [os.path.basename(c) for c in checkpoints]
+    return set(names)
+
+
+def main(megatron_cfg_path, checkpoint_path):
+    ppo_config = trl_config.method
+
+    megatron_cfg = OmegaConf.load(megatron_cfg_path)
+    megatron_cfg.trainer.num_nodes = 1
+    megatron_cfg.trainer.devices = 4
```

Comment:
For me inference fails with a single process, unless I manually set global batch size here, like so:
```python
megatron_cfg.model.global_batch_size = 64
```
But it also fails for me with multiprocessing, running `srun --label $PY -u /mnt/nvme/home/reciprocate/trlx/examples/nemo_ppo_inference.py configs/nemo_configs/megatron_1.3b.yaml nemo_1.3B_ppo_sentiments
` over `ntasks-per-node=8` with
```
4: │ ❱ 79 │   main(sys.argv[1], sys.argv[2])                                      │                                                                                                                                              [198/1868]
4: │   80                                                                         │
4: │                                                                              │
4: │ /mnt/nvme/home/reciprocate/trlx/examples/nemo_ppo_inference.py:61 in main    │
4: │                                                                              │
4: │   58 │   │   app_state.data_parallel_size,                                   │
4: │   59 │   │   app_state.pipeline_model_parallel_split_rank,                   │
4: │   60 │   │   app_state.virtual_pipeline_model_parallel_rank,                 │
4: │ ❱ 61 │   ) = fake_initialize_model_parallel(                                 │
4: │   62 │   │   world_size=app_state.model_parallel_size,                       │
4: │   63 │   │   rank=trainer.global_rank,                                       │
4: │   64 │   │   tensor_model_parallel_size_=megatron_cfg.model.tensor_model_par │
4: │                                                                              │
4: │ /mnt/nvme/home/reciprocate/miniconda3/envs/nemo/lib/python3.8/site-packages/ │
4: │ nemo/collections/nlp/modules/common/megatron/megatron_init.py:225 in         │
4: │ fake_initialize_model_parallel                                               │
4: │                                                                              │
4: │   222 │   │   │   │   data_parallel_group = list(ranks)                      │
4: │   223 │   │   │   │   logging.info(f'Rank {rank} has data parallel group: {d │
4: │   224 │                                                                      │
4: │ ❱ 225 │   data_parallel_rank = data_parallel_group.index(rank)               │
4: │   226 │   logging.info(f'All data parallel group ranks: {all_data_parallel_g │
4: │   227 │   logging.info(f'Ranks {rank} has data parallel rank: {data_parallel │
4: │   228                                                                        │
4: ╰──────────────────────────────────────────────────────────────────────────────╯
4: UnboundLocalError: local variable 'data_parallel_group' referenced before
4: assignment
srun: error: cw-prod-a100-cu117-40: tasks 1-3,5-7: Exited with exit code 1

Reply:
oh okay ill try to repro

Reply:
Should be fixed now. Was due to 1.3B being tp=1 pp=1 and the inference assumes num_gpus = tp*pp (since it supports only dp world = 1)

Reply:
not sure why the global batch size thing is happening but automatically setting it to mbs seems to work fine